pkg/workceptor/workceptor.go

```diff
@@ -678,25 +688,91 @@ func (w *Workceptor) unitStatusForCFR(unitID string) (map[string]interface{}, er
 	return retMap, nil
 }
 
+func (w *Workceptor) attemptRemoteAction(remoteNodeID, remoteUnitID, remoteAction string, unit *workUnit) error {
+	timeout := time.After(300 * time.Second)
+	retryInterval := 500 * time.Millisecond
+	shouldTimeout := func() bool {
+		select {
+		case <-timeout:
+			logger.Error("%s of remote work unit timed out", remoteAction)
+			return true
+		default:
+			time.Sleep(retryInterval)
```

Comment:
Instead of doing a `time.Sleep` in the `default` section, it would be better to do a `case <- time.After(retryInterval)`